iceoryx_binding_c/include/iceoryx_binding_c/internal/c2cpp_binding.h

```diff
@@ -21,16 +21,16 @@
 
 #include <cstdint>
 
-#define CLASS class
-#define ENUM
+#define IOX_CLASS class
+#define IOX_ENUM
 
 #else
 
 #include <stdbool.h>
 #include <stdint.h>
 
-#define CLASS struct
-#define ENUM enum
+#define IOX_CLASS struct
+#define IOX_ENUM enum
 
 #endif
```

Comment:
I'm not even sure why this is required?

Can you remove this part and the corresponding `#endif`
```cpp
#ifdef __cplusplus

#include <cstdint>

#define CLASS class
#define ENUM

#else
```

We can then run the CI and check if something breaks but I'm quite sure that all of our CI targets should still run. If this is the case, we can remove the defines completely

Reply:
It looks like it's the difference between treating the code as C++ versus C.  Do you have parts of your code that need to be compiled in both forms?

In C, if you have an enum, you need to use the `enum` keyword to create an instance, since it's part of the type.

```c
enum Food {
   Apple,
   Cherry
};

enum Food food = Apple;
```

But in C++, the `enum` keyword is optional when specifying the type.

As for `CLASS`, C doesn't have the `class` keyword, only `struct`.  In C++, the `struct` keyword implies public default visibility instead of private default visibility with `class`.

If you no longer have the requirement to compile your headers for both C and C++, then yeah, we can probably get rid of these.

Reply:
Enum example in C++: https://godbolt.org/z/P8shnz18j
Enum example in C: https://godbolt.org/z/bc3jMeavG

Reply:
Looking closer, I think you're right.  Whether interpreted in C or C++, the

```
#define CLASS struct
#define ENUM enum
```

should work based on how it's used.  The `CLASS` macro is used only to make typedefs.  The `ENUM` macro is only for types.  I'll give it a try, thanks! 👍

Reply:
@elBoberido if everything looks good, I'm happy to squash my commits into a single commit and update the PR title and description.

Reply:
Okay, now I understand why this was done. I'm not sure if it was a good idea but there are typedefs with some C++ classees, e.g.
```C
typedef struct UntypedClient* iox_client_t;
```
The `UntypedClient` is actually a C++ class and if the header is used in a C++ context it needs to be just `typedef UntypedClient* iox_client_t`.

I'm not quite sure why the compiler does not complain since `UntypedClient` is in a C++ namespace.

In order to get rid of the defines, one would need to create something like this
```C
struct iox_client;
typedef struct iox_client* iox_client_t;
```

And the always do a reinterpret_cast between the `iox_client_t` and the `UntypedClient*`.

I would suggest to keep the current approach but name the define `IOX_C_ENUM` and `IOX_C_CLASS`.

Reply:
I tested and found that it's fine on Linux using GCC.  I didn't know about problems with the Microsoft ABI:

> error: struct 'UntypedClient' was previously declared as a class; this is valid, but may result in linker errors under the Microsoft C++ ABI

That's good to know.  The `enum` should be totally fine, right?  AFAIK, the `enum` keyword in C++ for types are optional, but are required in C.

```c++
enum Food {
  Apple,
  Cherry
};

const char* foodToString(enum Food food) {
  switch(food) {
    case Apple: return "Apple";
    case Cherry: return "Cherry";
  }
  return "??";
}
```

The above is valid C code, and valid C++ code.  If we removed the `enum` from `const char* foodToString(enum Food food)`, then it's no longer valid C code.

Reply:
I'm not sure if there are some compiler quirks on some platforms. You can try it though.

Reply:
I( pushed a version that removes `ENUM` and renames `CLASS`.  Let's see if that passes.

Reply:
It worked on all platforms with any compiler.

Once you squash a few commits, this can be merged.

Reply:
I merged all commits into one, and added the to the copyright headers of modified files.

Thanks!